server/routes/v1/upload.ts

```diff
@@ -192,15 +193,43 @@ export const postUpload = [
       ).add({
         versionId: version._id,
         userId: req.user?._id,
-        binary: normalizeMulterFile(files.binary[0]),
-        code: normalizeMulterFile(files.code[0]),
+        binary: normalizeMulterFile(files.binary[0], 'binary', version),
+        code: normalizeMulterFile(files.code[0], 'code', version),
       })
+
       req.log.info({ code: 'created_upload_job', jobId }, 'Successfully created job in upload queue')
 
       // then return reference to user
       res.json({
         uuid: model.uuid,
       })
+
+      try {
+        const binaryLocation = `model/${model._id}/version/${version._id}/raw/binary/${files.binary[0].path}`
+        const client = new Minio.Client(config.get('minio'))
+        await client.copyObject(
+          files.binary[0].bucket,
+          binaryLocation,
+          `${files.binary[0].bucket}/${files.binary[0].path}`,
+          new Minio.CopyConditions()
+        )
+        await client.removeObject(files.binary[0].bucket, files.binary[0].path)
+
+        const codeLocation = `model/${model._id}/version/${version._id}/raw/code/${files.code[0].path}`
+        await client.copyObject(
+          files.code[0].bucket,
+          codeLocation,
+          `${files.code[0].bucket}/${files.code[0].path}`,
+          new Minio.CopyConditions()
+        )
+        await client.removeObject(files.code[0].bucket, files.code[0].path)
+
+        version.rawBinaryPath = binaryLocation
+        version.rawCodePath = codeLocation
+        version.save()
```

Comment:
`.save()` is an async function and needs to be run with `await`.  I wonder if that led to some of the random issues you were seeing before.